cpu/msp430fxyz/periph/spi.c

```diff
@@ -210,18 +211,33 @@ int spi_release(spi_t dev)
     return 0;
 }
 
-int spi_transfer_byte(spi_t dev, char out, char *in)
+int spi_transfer_bytes(spi_t dev, char *out, char *in, unsigned int length)
 {
     (void)dev;
-    char tmp;
-    while (!(SPI_IF & SPI_IE_TX_BIT)) {}
-    SPI_DEV->TXBUF = (uint8_t)out;
-    while (!(SPI_IF & SPI_IE_RX_BIT)) {}
-    tmp = (char)SPI_DEV->RXBUF;
-    if (in) {
-        *in = tmp;
+
+    assert(out || in);
+
+    /* if we only send out data, we do this the fast way... */
+    if (!in) {
+        for (unsigned i = 0; i < length; i++) {
+            while (!(SPI_IF & SPI_IE_TX_BIT)) {}
+            SPI_DEV->TXBUF = (uint8_t)out[i];
+            while (SPI_DEV->STAT & USCI_SPI_STAT_UCBUSY) {}
+            SPI_DEV->RXBUF;
```

Comment:
Both lines should go below the closing brace in line 227 (just be called once, after the transmission of multiple bytes). However, we will need to adapt this passage for the older msp430 revisions so we can do it then.